### PR TITLE
Add & remove task

### PR DESCRIPTION
**Pull request by ridabensalem**
in this milestone of to-do list project we have :   

-  Remove all hardcoded items from the tasks array.
-  Create a new JavaScript file for the new functionality.🔨
-  Implement a function for adding a new task (add a new element to the array).:heavy_plus_sign:
-  Implement a function for deleting a task (remove an element from the array).:heavy_minus_sign:
-  Implement a function for editing task descriptions.:pencil2:

### CODE REVIEW

**Review by rloterh — COMMENTED** [review]
Hi Rida, 

While you made a great effort in this project, unfortunately, I cannot proceed to review your code.

## Invalid Code Review Request

Kindly ensure that your ReadMe for this project adhere(s) to the following standard(s):
- [ ] All main headers from the [ReadMe template](https://github.com/microverseinc/readme-template) should be present in your ReadMe, only the FAQ section is optional.
- [ ]  The _Future Features_ section of your ReadMe includes 1-3 potential features to be added later. 
- [ ] Kindly ensure to remove the Microverse template logo. Optionally, you can replace it with your own logo if you have one.
- [ ] Kindly ensure that provided text instructions are removed and replaced with appropriate contents applicable under each section of your ReadMe.

Your Code Review Request will be marked as invalid in your Dashboard, so please submit a new one once you are ready 🙏

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

--------

_Invalid Code Review Request **does not count** into the code reviews limit._

**Review by m05e5 — CHANGES_REQUESTED** [review]
Hi @ridabensalem , 

Good job so far. !
There are some issues that you still need to work on to go to the next project but you are almost there!
mario_ladders


## Highlights


## Required Changes ♻️
* You really did a great job but there are some features you need to improve

- [ ] Hi as part of the requirements all changes should be stored in local storage. So, kindly save the edited todos in local storage too

_I'm sure you can do it just keep it up_

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

_@m05e5_

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/modules/tasksApp.js [hunk 1]
```diff
@@ -51,12 +53,9 @@ const tasksApp = () => {
 
   // Retrieve the listArray from Local Storage when the page is loaded..............
```
Comment (omarsalem7) [inline src/modules/tasksApp.js:60]: - [ ] After you made the request changes stated by the previous reviewer, there is another bug which after adding new todos, appears as undefined but it is saved in local storage. Also, I can't edit the todo, kindly fix it also. 
>image

**Review by omarsalem7 — CHANGES_REQUESTED** [review]
Hi @ridabensalem 😊, 

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't been addressed yet.

Almost near

 ## Highlights
✔️ There is no linter error present
✔️ The Pull Request has a proper title and description

 ## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
- N/A

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/modules/tasksApp.js (hunk 1 above)
Comment (ridabensalem) [reply src/modules/tasksApp.js:60]: Please Note that you can't edit until you select the task in the three dots listed on the left this is one of the requirements in the Big picture about the project mentioned in the following video: 
[Big picture about the project ](https://www.youtube.com/watch?v=AcUd-_Yjjqg&ab_channel=EmiliaAndrzejewska)
Note : please for the next reviewer watch this video before reviewing

**Review by MahmoudBakr23 — APPROVED** [review]
Hi @ridabensalem 😃, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit:
Congratulations! 🎉 

**Highlights**
- Well done on making the required changes from the previous reviewer.

Cheers and Happy coding!👏👏👏

Feel free to mention @MahmoudBakr23 to leave any questions or comments in the PR thread if something is not 100% clear.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**